src/setup_node_env/babel_register/register.js

```diff
@@ -41,9 +41,8 @@ var ignore = [
   // is `x-pack` and `b` is not `node_modules`
   /\/node_modules\/(?!x-pack\/(?!node_modules)([^\/]+))([^\/]+\/[^\/]+)/,
 
-  // ignore paths matching `/canvas/canvas_plugin/{a}/{b}` unless
-  // `a` is `functions` and `b` is `server`
-  /\/canvas\/canvas_plugin\/(?!functions\/server)([^\/]+\/[^\/]+)/,
+  // ignore paths matching `/canvas/canvas_plugin/`
+  /\/canvas\/canvas_plugin\//,
```

Comment:
By sharing the server_code_transformer with `@kbn/interpreter` I was able to pre-transpile the server code in the canvas_plugin_src and remove this customization.